### PR TITLE
Intel 2022 updates

### LINKED ISSUE #623: Bug fix & safeguard updates
Two GSI-specific issues were identified while testing `develop` at [008c63c](https://github.com/NOAA-EMC/GSI/commit/008c63cc04d6d80c25b0c3220b2d2c3f98618d52) in the global-workflow.
1. bug:  `gsi.x` encodes `Time` into netcdf radiance diagnostic file but `enkf.x` expects `Obs_Time`
2. safeguard:  PR #571 modified how `gsi.x` creates sub-directories.  This update did not consider the case in which the directory to be created already exists.

This issue is opened to address both of these points.

### PR DESCRIPTION
**Pull request by RussTreadon-NOAA**
**Description**
This PR fixes two types of bugs discovered when cycling `gsi.x` and `enkf.x` with intel/2022 in the global workflow
1. modify variables written to netcdf diagnostic files by `gsi.x` to be consistent with codes which read netcdf diagnostic files
2. modify `lrun_subdirs=.true.` option of `gsi.x` to properly handle the case in which sub-directories already exist in the run directory

Fixes #623

**Type of change**
- [x] Bug fix (non-breaking change which fixes an issue)

**How Has This Been Tested?**
Ctests have been on Hera, Orion, and WCOSS2 (Cactus) with acceptable behavior.  A global parallel covering the period 2021073106 through 2021080118 has been run on Hera, Orion, and WCOSS2 (Cactus).  All global workflow jobs ran as expected.
  
**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] New and existing tests pass with my changes

### CODE REVIEW

File: src/gsi/obsmod.F90 [hunk 1]
```diff
@@ -1038,10 +1038,15 @@ subroutine init_directories(in_pe,num_pe)
...
-             if(.not. l_mkdir_stat) then
-                write(6, *) "Failed to create directory ", trim(loc_dirname), " for PE ", loc_pe_name
-                call stop2(678)
+             INQUIRE(directory=trim(loc_dirname), exist=l_dir_exist)
+             if (.not.l_dir_exist) then
+                l_mkdir_stat = MAKEDIRQQ(trim(loc_dirname))
+                if(.not.l_mkdir_stat) then
+                   ierror=GETLASTERRORQQ()
+                   write(6, *) "INIT_DIRECTORIES:  ***ERROR** Failed to create directory ", &
+                        trim(loc_dirname)," for PE ", loc_pe_name, ' ierror= '
```
Comment (DavidHuber-NOAA) [inline src/gsi/obsmod.F90]: Missing the variable `ierror` in the write statement.
```suggestion
                        trim(loc_dirname)," for PE ", loc_pe_name, ' ierror= ', ierror
```

**Review by DavidHuber-NOAA — CHANGES_REQUESTED** [review]
Looks good.  I have tested these code changes in my own cycled test as well and they seem to work.  I have just one minor change request.

File: src/gsi/obsmod.F90 (hunk 1 above)
Comment (RussTreadon-NOAA) [reply src/gsi/obsmod.F90]: Thank you, @DavidHuber-NOAA , for catching my mistake.  What's the point of capturing an error flag if you don't actually print the value of the flag when an error occurs.   Bug corrected at 2bc4f13f.

**Review by DavidHuber-NOAA — APPROVED** [review]

**Comment by RussTreadon-NOAA** [comment]
Thanks @DavidHuber-NOAA

**Comment by CoryMartin-NOAA** [comment]
@RussTreadon-NOAA I'll handle merging this for you as the 'handling reviewer'. Do you wish to wait for @CatherineThomas-NOAA to approve, or will my review + @DavidHuber-NOAA 's be sufficient?

**Review by CoryMartin-NOAA — APPROVED** [review]

**Review by CatherineThomas-NOAA — APPROVED** [review]
Looks good to me

**Pull request merged**